magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,155 @@
+	return ExecuteTestAction(rctx)
+}
+
+// InfraDeploymentsTestFilesOnlyRule defines rules of test suites running of each changed component
+var InfraDeploymentsTestFilesOnlyRule = rulesengine.Rule{Name: "Infra-deployments PR Components File Diff Execution",
+	Description: "Runs specific tests of changed component by infra-deployments PR.",
+	Condition: rulesengine.Any{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule},
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		// Adding konflux-demo to the label filtetr when componenets changed
+		if rctx.LabelFilter == "" {
+			rctx.LabelFilter = "konflux"
+		} else {
+			rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "konflux")
+		}
+		return nil
+
+	}),
+		rulesengine.ActionFunc(ExecuteTestAction)}}
+
+var InfraDeploymentsIntegrationTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Integration component File Change Rule",
+	Description: "Map Integration tests files when Integration component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/integration/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "integration-service")
+		return nil
+	})}}
+
+var InfraDeploymentsEnterpriseControllerTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Enterprise Controller component File Change Rule",
+	Description: "Map Enterprise Controller tests files when EC component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/enterprise-contract/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "ec")
+		return nil
+	})}}
+
+var InfraDeploymentsJVMTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Jvm-build-service component File Change Rule",
+	Description: "Map jvm-build-service tests files when Jvm-build-service component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/jvm-build-service/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "jvm-build-service")
```

Comment:
since [jvm-build-service tests are skipped](https://github.com/konflux-ci/e2e-tests/blob/7eb9fd8575f3c52d76d4d2736f92ff6a5038e3bb/tests/build/jvm-build.go#L39), will it be good to skip the jobs instead [here](https://github.com/openshift/release/blob/4a1405501f3ee3406a34fcbe4b4f9bb8d58f6c5b/ci-operator/config/redhat-appstudio/infra-deployments/redhat-appstudio-infra-deployments-main.yaml#L47) ito avoid provisioning cluster?

Reply:
@tisutisu we still run jbs test in ["konflux" suite](https://github.com/konflux-ci/e2e-tests/blob/7eb9fd8575f3c52d76d4d2736f92ff6a5038e3bb/tests/konflux-demo/konflux-demo.go#L384) - should we disable that one as well?

Reply:
marked [it as pending](https://github.com/konflux-ci/e2e-tests/blob/7eb9fd8575f3c52d76d4d2736f92ff6a5038e3bb/tests/konflux-demo/konflux-demo.go#L386) as well?